be/src/storage/lake/rowset.cpp

```diff
@@ -67,6 +90,33 @@ Rowset::~Rowset() {
     }
 }
 
+void Rowset::add_uncompacted_segments(TxnLogPB_OpCompaction* op_compaction, uint64_t& uncompacted_num_rows,
+                                      uint64_t& uncompacted_data_size) const {
+    if (!partial_segments_compaction()) { // defensive check
+        return;
+    }
+    bool has_encryption_meta = (metadata().segments_size() == metadata().segment_encryption_metas_size());
+    for (size_t i = _compaction_segment_limit, idx = 0; i < metadata().segments_size(); ++i, ++idx) {
+        op_compaction->mutable_output_rowset()->add_segments(metadata().segments(i));
+        StatusOr<int64_t> size_or = _uncompacted_segments[idx]->get_data_size();
+        int64_t file_size = 0;
+        if (size_or.ok()) {
+            file_size = *size_or;
+            op_compaction->mutable_output_rowset()->add_segment_size(file_size);
+        } else {
+            LOG(WARNING) << "unable to get file size for segment " << metadata().segments(i) << " for tablet "
```

Comment:
what's the error? shall be included in to warning log as well.

Reply:
done